sdcm/utils/issues.py

```diff
@@ -0,0 +1,87 @@
+import re
+import sys
+import logging
+from functools import lru_cache
+from dataclasses import dataclass
+
+import github
+import github.Auth
+from github.GithubException import UnknownObjectException
+
+from sdcm.keystore import KeyStore
+from sdcm.sct_config import SCTConfiguration
+from sdcm.sct_events.base import Severity
+from sdcm.sct_events.system import TestFrameworkEvent
+
+scylla_issue_pattern = re.compile(
+    r"^\s*((((?P<user_id>[\w:-]+)/)?(?P<repo_id>[\w:-]+))?#)?(?P<issue_id>\d+)\s*$", re.IGNORECASE)
+
+scylla_issue_https_pattern = re.compile(
+    r"^https?:/.*?/(?P<user_id>[\w:-]+)/(?P<repo_id>[\w:-]+)/.*?/(?P<issue_id>\d+?)$", re.IGNORECASE)
+
+
+@dataclass
+class Issue:
+    user_id: str | None
+    repo_id: str | None
+    issue_id: int | None
+
+
+class SkipPerIssues:
+    def __init__(self, issues: list[str] | str, params: SCTConfiguration | dict):
+        self.params = params
+
+        github_access = KeyStore().get_json("github_access.json")
+        auth = github.Auth.Token(token=github_access["token"])
+        self.github = github.Github(auth=auth)
+
+        issues = [issues] if isinstance(issues, str) else issues
+
+        self.issues = [self.get_issue_details(issue) for issue in issues]
+        self.issues = list(filter(lambda x:  x, self.issues))  # filter None - unmatched issues
+
+    @staticmethod
+    def parse_issue(issue_str):
+        for pattern in [scylla_issue_pattern, scylla_issue_https_pattern]:
+            match = pattern.search(issue_str.strip())
+            if match:
+                obj = match.groupdict()
+                return Issue(
+                    user_id=obj.get("user_id") or "scylladb",
+                    repo_id=obj.get("repo_id") or "scylladb",
+                    issue_id=int(obj.get("issue_id"))
+                )
+        return None
+
+    @lru_cache
+    def get_issue_details(self, issue):
+        issue_parsed = self.parse_issue(issue)
+        if not issue_parsed:
+            # TODO raise event about it
+            logging.warning("couldn't parse issue: %s", issue)
+            TestFrameworkEvent(source=self.__class__.__name__,
+                               message=f"couldn't parse issue: {issue}",
+                               severity=Severity.WARNING,
+                               trace=sys._getframe().f_back).publish()  # pylint: disable=protected-access
+            return None
+        try:
+            return self.github.get_user(issue_parsed.user_id).get_repo(issue_parsed.repo_id).get_issue(issue_parsed.issue_id)
+        except UnknownObjectException:
```

Comment:
in case of any GH issue (e.g. unavailable) it will raise an unhandled exception failing the nemesis.
One option would be to raise "UnsupportedNemesis" to mark nemesis as skipped in that case.

Reply:
This won't be used only for nemesis, so we can't assume that the exception we'll want